libtransmission/peer-io.cc

```diff
@@ -130,7 +130,15 @@ std::shared_ptr<tr_peerIo> tr_peerIo::new_outgoing(
 
     auto peer_io = tr_peerIo::create(session, parent, &info_hash, false, is_seed);
 
+    // try a TCP socket
+    if (auto sock = tr_netOpenPeerSocket(session, addr, port, is_seed); sock.is_valid())
```

Comment:
Will UTP ever be tried in this case?  Let's say TCP socket is opened, but times out.  It will just keep returning here (unless the atom is removed), even though the utp=true param is set by the caller. The connect() call is nonblocking of course, so this will always return.

The Tr3 logic was to try UTP first, let it timeout, market the atom as utp-failed, and return here from initiateConnection() and try to TCP connection. But with this change, it seems like it will never get to try UTP?

Reply:
@ckerr any comment on that?

Reply:
@reardonia you've got a point, but before this PR we were never getting a timeout callback from utp either, so there's some other issue going on here. I need to test further and figure it out, but thanks very much for the detective work :+1: